### PR TITLE
Merge master into features

### PR DESCRIPTION
**Pull request by nicoddemus**

### CODE REVIEW

File: src/_pytest/compat.py [hunk 1]
```diff
@@ -251,7 +257,7 @@ def get_real_method(obj, holder):
     try:
         is_method = hasattr(obj, "__func__")
         obj = get_real_func(obj)
-    except Exception:
+    except Exception:  # pragma: no cover
```
Comment (blueyed) [inline src/_pytest/compat.py]: wut?
Rather leave it show up in reports..

**Comment by nicoddemus** [comment]
Thanks! Please rebase #5747 on `features` as I plan to release 5.1.0 right away. 👍

**Comment by blueyed** [comment]
You can/should release regardless of #5747, no?

**Comment by nicoddemus** [comment]
Yep, only later I saw it closely and noticed it did not affect the release in the end. 👍

**Pull request merged**